src/modules/45-dashboards/pages/home/DashboardForm.tsx

```diff
@@ -51,106 +50,81 @@ export interface DashboardFormikValues {
+    const destinationFolderId = mode === 'EDIT' ? initialFolderId : firstEditableFolder
     return {
-      folderId: selectedFolderId,
+      folderId: destinationFolderId,
       name: formData?.name || '',
       description: formData?.description?.split(TAGS_SEPARATOR) || []
     }
-  }, [folderId, folderListItems, formData?.description, formData?.folderId, formData?.name])
+  }, [editableFolders, folderId, folderListItems, formData?.description, formData?.folderId, formData?.name])
 
   return (
     <Layout.Vertical padding="xxlarge" spacing="large">
       <Heading level={3} font={{ variation: FontVariation.H3 }}>
         {title}
       </Heading>
       <ModalErrorHandler bind={setModalErrorHandler} />
-      {isFoldersLoading ? (
-        <PageSpinner />
-      ) : (
-        <Formik
-          initialValues={initialValues}
-          enableReinitialize
-          formName="dashboardForm"
-          validationSchema={Yup.object().shape({
-            folderId: Yup.string().trim().required(getString('dashboards.folderForm.folderNameValidation')),
-            name: Yup.string().trim().required(getString('dashboards.createModal.nameValidation'))
-          })}
-          onSubmit={(completedFormData: DashboardFormikValues) => {
-            onComplete({
-              ...completedFormData,
-              description: completedFormData.description?.join(TAGS_SEPARATOR) || ''
-            })
-          }}
-        >
-          <Form className={css.formContainer}>
-            <Layout.Vertical spacing="large">
-              <FormInput.Select
-                name="folderId"
-                items={folderListItems}
-                label={getString('dashboards.homePage.folder')}
-                placeholder={getString('dashboards.resourceModal.folders')}
-              />
-              <FormInput.Text
-                name="name"
-                label={getString('name')}
-                placeholder={getString('dashboards.createModal.namePlaceholder')}
-              />
-              <FormInput.KVTagInput name="description" label={getString('tagsLabel')} isArray={true} />
-              <Button
-                className={css.button}
-                disabled={loading}
-                intent="primary"
-                text={getString('continue')}
-                type="submit"
-                variation={ButtonVariation.PRIMARY}
-              />
-            </Layout.Vertical>
-          </Form>
-        </Formik>
-      )}
+      <Formik
+        initialValues={initialValues}
+        enableReinitialize
+        formName="dashboardForm"
+        validationSchema={Yup.object().shape({
+          folderId: Yup.string().trim().required(getString('dashboards.folderForm.folderNameValidation')),
+          name: Yup.string().trim().required(getString('dashboards.createModal.nameValidation'))
+        })}
+        onSubmit={(completedFormData: DashboardFormikValues) => {
```

Comment:
Nothing wrong here, but you can actually type your form and it'll folder though to things like `onSubmit`
```suggestion
      <Formik<DashboardFormikValues>
        initialValues={initialValues}
        enableReinitialize
        formName="dashboardForm"
        validationSchema={Yup.object().shape({
          folderId: Yup.string().trim().required(getString('dashboards.folderForm.folderNameValidation')),
          name: Yup.string().trim().required(getString('dashboards.createModal.nameValidation'))
        })}
        onSubmit={completedFormData => {
```

Though it can then get a bit snippy about some things, so your mileage may vary